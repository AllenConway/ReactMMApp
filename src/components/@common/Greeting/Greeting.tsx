import React, { useState } from "react";
import { useEffect } from "react";

interface GreetingProps {
  greetingMessage?: string;
}

const Greeting: React.FC<GreetingProps> = (props: GreetingProps) => {
  //use React hooks
  const [greetingMessage, setGreetingMessage] = useState(
    "Welcome to Mountain Cabin Vacations"
  );

  useEffect(() => {
    if (props.greetingMessage) {
      setGreetingMessage(props.greetingMessage);
    }
  });

  return <h1>{greetingMessage}</h1>;
};

export default Greeting;
