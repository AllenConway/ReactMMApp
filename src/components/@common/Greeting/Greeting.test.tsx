import React from 'react';
import { create } from "react-test-renderer";
import Greeting from './Greeting';

describe("Grreting component", () => {
    test("Matches the snapshot", () => {
      const greeting = create(<Greeting />);
      expect(greeting.toJSON()).toMatchSnapshot();
    });
});