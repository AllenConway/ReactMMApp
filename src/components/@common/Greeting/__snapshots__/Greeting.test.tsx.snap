// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grreting component Matches the snapshot 1`] = `
<h1>
  Welcome to Mountain Cabin Vacations
</h1>
`;
